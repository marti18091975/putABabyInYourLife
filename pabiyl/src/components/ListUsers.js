import React, { useState, useEffect } from "react";
//import { Prompt } from 'react-router-dom';
//import user from '../stores/UserStore';
import userStore from "../stores/UserStore";
import { loadDetails } from "../actions/detailAction";
import { NavLink } from 'react-router-dom';

import "./listUsers.css";

function ListUsers(props) {

    const [users, setUsers] = useState([]);
    const actualUsersList = users.map((user) => {
        const link = '/detailUser/' + user._id;
        return (

            <div className="user-list__item">
                <img className="list-image" alt="" src={user.mainImage}></img>
                <div className="right-side">
                    <div className="text name">{user.name}</div>
                    <div className="right-side__container">
                        <div className="box">
                            <img alt="" className="img__description" src="https://image.flaticon.com/icons/svg/888/888063.svg" ></img>
                            <div className="text">{user.city}</div>
                        </div>
                        <div className="box">
                            <img alt="" className="img__description" src="https://image.flaticon.com/icons/svg/3021/3021790.svg"></img>
                            <div className="text">{user.age}</div>
                        </div>
                    </div>
                    <div className="right-side__container">
                        <div className="box">
                            <img alt="" className="img__description" src="https://image.flaticon.com/icons/svg/1761/1761430.svg"></img>
                            <div className="text">{user.sons} sons</div>
                        </div>
                        <div className="box">
                            <img alt="" className="img__description" src="https://image.flaticon.com/icons/svg/1102/1102457.svg" ></img>
                            <div className="text">{user.civilState}</div>
                        </div>
                    </div>
                </div>

            </div>

        );
    });
    useEffect(() => {
        userStore.addChangeListener(onChange);

        loadDetails();

        return () => userStore.removeChangeListener(onChange);
    });
    function onChange() {
        setUsers(userStore.getDetailUsers());
    }
    return (
        <div className="main__list">

            <ul className="user-list">{actualUsersList}</ul>

        </div>
    );


}
export default ListUsers;