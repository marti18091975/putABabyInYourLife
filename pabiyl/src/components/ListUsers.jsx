import React from "react";
import { NavLink } from "react-router-dom";
import "./listUsers.scss";

function ListUsers(props) {
  const actualUsersList = props.usersList.map((user) => {
    const link = "/detailUser/" + user._id;
    return (
      <NavLink className="nav-link" key={user._id} to={link}>
        <div className="user-list__item">
          <img className="list-image" alt="" src={user.mainImage}></img>
          <div className="right-side">
            <div className="text name">{user.name}</div>
            <div className="right-side__container">
              <div className="box">
                <img
                  alt=""
                  className="img__description"
                  src="https://image.flaticon.com/icons/svg/888/888063.svg"
                ></img>
                <div className="text">{user.city}</div>
              </div>
              <div className="box">
                <img
                  alt=""
                  className="img__description"
                  src="https://image.flaticon.com/icons/svg/3021/3021790.svg"
                ></img>
                <div className="text">{user.age}</div>
              </div>
            </div>
            <div className="right-side__container">
              <div className="box">
                <img
                  alt=""
                  className="img__description"
                  src="https://image.flaticon.com/icons/svg/1761/1761430.svg"
                ></img>
                <div className="text">{user.sons} sons</div>
              </div>
              <div className="box">
                <img
                  alt=""
                  className="img__description"
                  src="https://image.flaticon.com/icons/svg/1102/1102457.svg"
                ></img>
                <div className="text">{user.civilState}</div>
              </div>
            </div>
          </div>
        </div>
      </NavLink>
    );
  });
  return (
    <div className="main__list">
      <ul className="user-list">{actualUsersList}</ul>
    </div>
  );
}
export default ListUsers;
