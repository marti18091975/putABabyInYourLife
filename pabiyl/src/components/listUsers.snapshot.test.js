import renderer from 'react-test-renderer';
import React from 'react';
import ListUsers from './ListUsers';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App snapshot', () => {
    const props = {
        usersList: []
    }

    const tree = renderer.create(
        <Router>
            <ListUsers {...props} />
        </Router>
    );

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
});