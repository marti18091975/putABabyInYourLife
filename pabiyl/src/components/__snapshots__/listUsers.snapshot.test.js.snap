// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<div
  className="main__list"
>
  <ul
    className="user-list"
  />
</div>
`;
